lib/ResolverFactory.js

```diff
@@ -31,12 +31,39 @@ const ResultPlugin = require("./ResultPlugin");
 const UnsafeCachePlugin = require("./UnsafeCachePlugin");
 const PnpPlugin = require("./PnpPlugin");
 
+/** @typedef {string|Array<string>|false} AliasOptionNewRequest */
+/** @typedef {{alias?: AliasOptionNewRequest, name?: string, onlyModule?: boolean}} AliasOptionEntry */
+/** @typedef {{[k: string]: AliasOptionNewRequest}} AliasOptions */
+
+/**
+ * @typedef {Object} ResolveOptions
+ * @property {(AliasOptions | Array<AliasOptionEntry>)=} alias
```

Comment:
Are we really ok with Closure compiler style? may be TypeScript `?` will be better?

Reply:
afaik this is the syntax for optional properties/parameters in jsdoc. `?` also works but is typescript specific and not official jsdoc